tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -85,11 +125,36 @@ def index_impl(self):
 
   def scalars_impl(self, tag, run, output_format):
     """Result of the form `(body, mime_type)`."""
-    tensor_events = self._multiplexer.Tensors(run, tag)
-    values = [[tensor_event.wall_time,
-               tensor_event.step,
-               tf.make_ndarray(tensor_event.tensor_proto).item()]
-              for tensor_event in tensor_events]
+    if self._db_connection_provider:
+      db = self._db_connection_provider()
+      cursor = db.execute('''
+        SELECT
+          Tensors.step,
+          Tensors.computed_time,
+          Tensors.data,
+          Tensors.dtype,
+        FROM Tensors
+        LEFT JOIN Tags
+        ON Tensors.series=Tags.tag_id
+        LEFT JOIN Runs
+        ON Tags.run_id=Runs.run_id '
+        WHERE
+          Runs.run_name = ?
+          AND Tags.tag_name = ?
+          AND Tags.plugin_name = ?
+          AND Tensors.shape = ''
+          AND Tensors.step > -1
```

Comment:
[Optional] Is there a tangible reason why we're doing this? Would UI fall apart if user had negative steps?

Reply:
Ah, this is indeed unclear. I added a comment. The check filters out placeholder rows.